### ISSUE TITLE
Adhere to Semantic Versioning

### ISSUE THREAD

**Issue by newnone** [claird/PyPDF4#16]
Probably the current versioning style is already close to it, but I propose to stick more seriously to the succinct specification of [Semantic Versioning (semver) 2.0.0](https://semver.org/). This will aid communicating the progress of the project to the external users of PyPDF.

A first problem that ensues is: where should PyPDF4' versioning begin? Since it is named somehow differently from its predecessor PyPDF2, shall it be treated as a new project and start at `0.1.0`? Or begin outright from `4.0.0`?

An advantage of `0.1.0` is that it communicates its non-public state, which PyPDF4 might be in. `4.0.0` has not that interpetation. On the other hand, PyPDFx has already been subject to long use and be considered in production, which `4.0.0` can adequately target.

**Comment by xilopaint** [comment]
`4.0.0` would make sense if PyPDF2 was `2.x.x` what is not the case since the last version is `1.26.0`. Given that I would stick with `0.1.0`.

Additionally, there's the weirdness that no one knows what happens with PyPDF3.

**Comment by xilopaint** [comment]
I'm not telling that anyone should be concerned about PyPDF3, I just told that adopting a `4.x.y` format is weird when PyPDF2 is `1.x.y` and no one know what happens with PyPDF3.

I personally dislike the fact this project couldn't be named PyPDF3 but the reasons behind this are not my business.

**Comment by newnone** [comment]
The restart to `0.1.0` is fine for me too. Let's hear what @claird et al. have to say :wink:.

**Comment by claird** [comment]
What I have to say in regard to 0.1.0 is ... oops:
https://pypi.org/project/PyPDF4/1.27.0/  Is there any realistic way we can
move from 1.27.0 to 0.1.0?

Cameron Laird, vice president
We make computers work for people.

**Comment by sim0nx** [comment]
> What I have to say in regard to 0.1.0 is ... oops:
https://pypi.org/project/PyPDF4/1.27.0/  Is there any realistic way we can
move from 1.27.0 to 0.1.0?

Hmm it is possible to delete an already published package on pypi. Though I don't know if they allow you to publish a package with a lower version number than the one you deleted. May try that and try publishing a 0.0.1 ?

**Comment by newnone** [comment]
>May try that and try publishing a 0.0.1 ?

If you do, please [start at 0.1.0](https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase). Patch versioning [only when introducing bug fixes](https://semver.org/#spec-item-6) and possibly other assimilated cases.

**Comment by sim0nx** [comment]
@newnone I suggested 0.0.1 just for testing... if it works, delete it again and publish the first version as 0.1.0 ... :-)

**Comment by kurtmckee** [comment]
Don't jump backwards to 0.1.0. This code builds on the contributions of 70 people already. It's not a reboot of the codebase.

Calling it 1.27.0 was the right decision when demonstrating PyPI uploads. Following semantic versioning will be a good decision. Jumping backwards in version numbers will lead to confusion about the stability of a codebase that has not been completely rewritten from scratch.

Move forward.

**Comment by newnone** [comment]
>Don't jump backwards to 0.1.0. This code builds on the contributions of 70 people already. It's not a reboot of the codebase.

I agree, and I've done so in [a past comment of "Package name is confusing"](https://github.com/claird/PyPDF4/issues/30) a few months ago, after consideration.

Amongst other things, the proposal also involves collpasing the number suffix `4` from the package name on PyPI, GitHub etc. One single PyPDF, with versions identified by the very version number.
